src/index.ts

```diff
@@ -0,0 +1,556 @@
+  gaugeUpdatedTime.set(executionLabels, new Date(execution._updated).getTime() / 1000);
+
+  gaugeExSlicers.set(executionLabels, execution.slicers);
+  gaugeExWorkers.set(executionLabels, execution.workers);
+  generateExecutionStatusMetrics(execution, executionLabels);
+}
+
+function generateControllerStats(terasliceStats:TerasliceStats, labels:any) {
+  // FIXME: I should rethink this warning
+  // eslint-disable-next-line no-restricted-syntax
+  for (const controller of terasliceStats.controllers) {
+    parseController(controller, labels);
+  }
+}
+
+function generateExecutionStats(terasliceStats:TerasliceStats, labels:any) {
+  // FIXME: I should rethink this warning
+  // eslint-disable-next-line no-restricted-syntax
+  for (const execution of terasliceStats.executions) {
+    parseExecution(execution, labels);
+  }
+}
+
+interface StateExecution {
+  exId: string,
+  jobId: string,
+  image: string
+}
+interface StateExecutionList {
+  [key: string]: StateExecution
+}
+
+/**
+ * NOTE: This assumes Teraslice is running in Kubernetes mode.
+ *
+ * generateExecutionVersions - takes the /cluster/state output, which looks like
+ * this:
+ *
+ *    "10.123.4.111": {
+ *      "node_id": "10.123.4.111",
+ *      "hostname": "10.123.4.111",
+ *      "pid": "N/A",
+ *      "node_version": "N/A",
+ *      "teraslice_version": "N/A",
+ *      "total": "N/A",
+ *      "state": "connected",
+ *      "available": "N/A",
+ *      "active": [
+ *          {
+ *              "assets": [],
+ *              "assignment": "worker",
+ *              "ex_id": "5ba1da6a-0ba2-49f4-92c3-d436ba510111",
+ *              "image": "teraslice:v0.70.0",
+ *              "job_id": "7e6dfa3c-6665-455d-9d52-f11bd32ad111",
+ *              "pod_name": "ts-wkr-my-job-name-1d940e75-58d9-74c54e7dc1-nxaaa",
+ *              "pod_ip": "10.132.86.111",
+ *              "worker_id": "ts-wkr-my-job-name-1d940e75-58d9-74c54e7dc1-nxaaa"
+ *          }
+ *        ]
+ *      }
+ *
+ * and makes something like this:
+ *
+ * {
+ *  exId: {exId, jobId, image},
+ *  exId: {exId, jobId, image}
+ * }
+ *
+ * That then gets used to generate the metrics.
+ *
+ * @param terasliceStats
+ * @param labels
+ */
+function generateExecutionVersions(terasliceStats:TerasliceStats, labels:any) {
+  const executions:StateExecutionList = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [, workerNode] of Object.entries(terasliceStats.state)) {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const worker of workerNode.active) {
+      if (worker.ex_id && !Object.prototype.hasOwnProperty.call(executions, worker.ex_id)) {
+        executions[worker.ex_id] = {
+          exId: worker.ex_id,
+          jobId: worker.job_id,
+          image: worker.image,
+        };
+      }
+    }
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [, execution] of Object.entries(executions)) {
+    const regex = /.*:(.*)_.*/g;
```

Comment:
What is this doing?

Reply:
Extracting what is assumed to be a version from an image like this:

```
terascope/teraslice:v0.71.1_XXXX
```

Now that you call it out ... that is an internal tag structure isn't it.  Ultimately, if it doesn't parse, I just want it to be an empty string.  I could probably massage that regex a bit so it's more flexible.

Oh, and to get `matchAll` to work, I had to set `"target": "esnext", ` in the `tsconfig.json`.